### PR TITLE
Rhys/sass flags

### PR DESCRIPTION
**Pull request by wheresrhys**
So this would work something like

```
@import n-ui/configure
@include nUiConfigure((preset: all, welcomeMessage: false))
@import n-ui/bootstrap
```
I forget exactly why I thought this would be a good idea. I think mainly because it looks more like the js api. Also allows a clearer separation between wanting a feature to be available, and silent mode flagging (which is more of an implementation detail that we may not want consumers to touch directly)

### CODE REVIEW

File: configure.scss [hunk 1]
```diff
@@ -0,0 +1,53 @@
+$_n-ui-features: (
+	ads: false,
+	buttons: false,
+	grid: false,
+	colors: false,
+	icons: false,
+	typography: false,
+	cookieMessage: false,
```
Comment (i-like-robots) [inline configure.scss]: I guess we should distinguish between 'foundations' (starting to dislike that already) and the non-core, opt-in 'components'. Have tried to define them a little in [the wiki](https://github.com/Financial-Times/n-ui/wiki).

Comment (i-like-robots) [inline configure.scss]: Oh yeah, and everything in Sass is lower case hyphenated - the only time I've seen camelCase in Sass is Origami. I wonder if we should pick a side.

Comment (wheresrhys) [inline configure.scss]: I'm on the fence as to whether the foundations/components distinction is worth having. Adds verbosity and, if we want to preserve the distinction at all levels of the API, complexity and demands on the user's memory.

Comment (i-like-robots) [inline configure.scss]: The reason I quite like sorting modules into folders is that there's no need to lookup and open files, it only takes a glance to see a directory structure - either locally or on Github.

I think the ability to quickly lookup what's bundled by default and what isn't is very useful but not wedded to any particular way of doing that.

**Comment by i-like-robots** [comment]
Just so it's recorded somewhere from our offline chat:

- expect configuration _before_ bootstrap is imported, throw `@error` if it is not
- output a `/*! loud */` comment displaying configuration info
- primitives (E.G. buttons, colors, icons) are not configurable

**Comment by wheresrhys** [comment]
... and in addition to the last point, if possible we should try and output a sensible subset of e.g. o-buttons, o-grid if origami allows

File: README.md [hunk 2]
```diff
@@ -1,50 +1,38 @@
...
+	if (flags.get('konami')) {
+		easterEgg.init();
+	}
+})
+```
 
-### Reduce duplication
+```Sass
+@import 'n-ui/configure';
+@include nUiConfigure((
```
Comment (i-like-robots) [inline README.md]: Should this be a function rather than a mixin as it has no output? Maybe print the config here?

File: configure.scss [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+	message-prompts: false,
```
Comment (i-like-robots) [inline configure.scss]: mix of camelCase and hyphenated keys

File: README.md (hunk 2 above)
Comment (wheresrhys) [inline README.md]: good call

File: configure.scss (hunk 3 above)
Comment (wheresrhys) [inline configure.scss]: Ah that's a mistake. I think we shoudl go camel case

File: configure.scss [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+
+	@if map-has-key($opts, preset) {
+		$preset: map-get($_n-ui-presets, map-get($opts, preset));
+		$_n-ui-features: map-merge($_n-ui-features, $preset) !global;
+		$opts: map-remove($opts, preset);
+	}
+
+	$_n-ui-features: map-merge($_n-ui-features, $opts) !global;
+
+	/* n-ui features: */
```
Comment (i-like-robots) [inline configure.scss]: Could make this a one liner maybe:

```scss
/*! n-ui features: #{inspect($_n-ui-features)} */
```
```

**Pull request merged**